src/misc/unix/SDL_sysurl.c

```diff
@@ -28,11 +28,25 @@
 #include <unistd.h>
 #include <sys/wait.h>
 #include <errno.h>
+#if USE_POSIX_SPAWN
+#include <spawn.h>
+extern char **environ;
+#endif
 
 int SDL_SYS_OpenURL(const char *url)
 {
     const pid_t pid1 = fork();
     if (pid1 == 0) { /* child process */
+#if USE_POSIX_SPAWN
+        pid_t pid2;
+        const char *args[] = { "xdg-open", url, NULL };
+        if (posix_spawnp(&pid2, args[0], NULL, NULL, (char **)args, environ) == 0) {
```

Comment:
Why is the (char **) cast needed?

Reply:
It's a deficiency in the POSIX prototype, which takes `char **` when it should have taken `char const * const *`.
Because `url` is a pointer to const char (as it should be) then either

1. `args[]` would need to be assigned `(char *)url`, or
2. The cast to `char **` in `posix_spawnp()`

Both options are gross, but that's POSIX...